Lib/test/test_threading.py

```diff
@@ -1338,6 +1339,21 @@ def run(self):
         # explicitly break the reference cycle to not leak a dangling thread
         thread.exc = None
 
+    def test_multithread_modify_file_noerror(self):
+        # See issue25872
+        def modify_file():
+            with open(__file__, 'a') as fp:
```

Comment:
`__file__` is not good here - it's the current file. Use a temp file from test.support.